core/rawdb/database_quorum.go

```diff
@@ -81,28 +86,39 @@ func GetPrivateBlockBloom(db ethdb.Database, number uint64) (bloom types.Bloom)
 	return bloom
 }
 
-type PrivacyMetadataLinker interface {
-	PrivacyMetadataRootForPrivateStateRoot(privateStateRoot common.Hash) common.Hash
-	LinkPrivacyMetadataRootToPrivateStateRoot(privateStateRoot, privacyMetadataRoot common.Hash) error
+// AccountExtraDataLinker maintains mapping between root hash of the state trie
+// and root hash of state.AccountExtraData trie
+type AccountExtraDataLinker interface {
+	// GetAccountExtraDataRoot returns the root hash of the state.AccountExtraData trie from
+	// the given root hash of the state trie.
+	//
+	// It returns an empty hash if not found.
+	GetAccountExtraDataRoot(stateRoot common.Hash) common.Hash
+	// Link saves the mapping between root hash of the state trie and
```

Comment:
1. Is the thrust of this change adding a state trie per contract account ?
2. Is it true that the node operator would unavoidably still see the plaintext of the transaction, but remote users of the node would never be able see transactions, receipts or logs in appropriate for their 'tenant' ?
3. Is the tenancy determined by the privateFor at contract deployment time (or something else) and can it be changed afterwords ?
4. Or, is there something else I can read that might help me understand the nature of these changes ?